homeassistant/components/imap_email_content/sensor.py

```diff
@@ -230,6 +230,12 @@ def update(self):
         email_message = self._email_reader.read_next()
 
         if email_message is None:
+            self._message = "unknown"
+            self._state_attributes = {
+                ATTR_FROM: "unknown",
```

Comment:
Our standard is to not add the item at all to state attributes if the value is unknown.